### PR TITLE
fix: Autocomplete doesn't show up in square brackets when the identifier has invalid characters

### LINKED ISSUE #13983: [Bug]-[266.5]:Autocomplete doesn't show up in square brackets when the identifier has invalid characters
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The best match for the auto complete doesn't show up in square brackets when the identifier has invalid characters.
[LOOM DEMO](https://www.loom.com/share/492f939257544bb2a641118c2ba0dc7f)

### Steps To Reproduce

1. Drag a table
2. Drag a text widget
3. Enter `{{Table1.selectedRow[' ']}}`.

**Expected:** Auto complete options should display in order to make it convenient for the user.
**Actual:** Auto complete `selectedRow` columns do not show up.

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Druthi**
## Description
Autocomplete doesn't show up in square brackets when the identifier has invalid characters

Fixes #13983 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Cypress

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [x] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts [hunk 1]
```diff
@@ -376,7 +376,7 @@ class CodeMirrorTernService {
           origins: true,
           caseInsensitive: true,
           guess: false,
-          inLiteral: false,
+          inLiteral: true,
```
Comment (ohansFavour) [inline app/client/src/utils/autocomplete/CodemirrorTernService.ts:379]: One caveat with setting this to `true` is that hints start appearing within literals. This might not be the expected behaviour.

Screenshot 2023-04-05 at 23 04 42
Screenshot 2023-04-05 at 23 10 15


We need to add a stringent condition in `handleAutocompleteKeydown` to ensure that autocompletion only shows up in literals within square brackets or an attempt to do so.

Comment (ohansFavour) [reply app/client/src/utils/autocomplete/CodemirrorTernService.ts:379]: Also, include jest/cypress test to account for this edge case (auto hints shouldn't appear within literals except inside a square bracket).

Comment (Druthi) [reply app/client/src/utils/autocomplete/CodemirrorTernService.ts:379]: The issue was, this list shows up only if the autocomplete menu is already visible, otherwise `handleAutocompleteKeydown` handles not showing this unless it is within `[""]`. I've pushed a fix for this and included tests.

**Review by ohansFavour — APPROVED** [review]

**Comment by ramsaptami** [comment]
LGTM 
cc: @Druthi
